File: tangem-card-old/src/main/java/com/tangem/tangem_card/util/Log.java
package com.tangem.tangem_card.util;

public class Log {
    public static void i(String logTag, String message) {
        if( loggerInstance!=null ) loggerInstance.i(logTag,message);
    }
    public static void e(String logTag, String message) {
        if( loggerInstance!=null ) loggerInstance.e(logTag,message);
    }
    public static void v(String logTag, String message) {
        if( loggerInstance!=null ) loggerInstance.v(logTag,message);
    }

    private static LoggerInterface loggerInstance=null;
    public static void setLogger(LoggerInterface logger)
    {
        loggerInstance=logger;
    }
}

File: tangem-card-new/src/main/java/com/tangem/Log.kt
package com.tangem

object Log {

    private var loggerInstance: LoggerInterface? = null

    fun i(logTag: String, message: String) {
        loggerInstance?.i(logTag, message)
    }

    fun e(logTag: String, message: String) {
        loggerInstance?.e(logTag, message)
    }

    fun v(logTag: String, message: String) {

        loggerInstance?.v(logTag, message)
    }

    fun setLogger(logger: LoggerInterface) {
        loggerInstance = logger
    }
}

interface LoggerInterface {
    fun i(logTag: String, message: String)
    fun e(logTag: String, message: String)
    fun v(logTag: String, message: String)
}